code/controllers/subsystems/ticker.dm

```diff
@@ -0,0 +1,537 @@
+	else
+		mode = config.pick_mode(GLOB.master_mode)
+
+	CHECK_TICK
+	if(!mode.can_start(bypass_checks))
+		to_chat(world, "Reverting to pre-game lobby.")
+		QDEL_NULL(mode)
+		job_master.ResetOccupations()
+		return FALSE
+
+	CHECK_TICK
+	if(!mode.pre_setup() && !bypass_checks)
+		QDEL_NULL(mode)
+		to_chat(world, "<b>Error in pre-setup for [GLOB.master_mode].</b> Reverting to pre-game lobby.")
+		job_master.ResetOccupations()
+		return FALSE
+
+	CHECK_TICK
+	if(!mode.setup() && !bypass_checks)
+		QDEL_NULL(mode)
+		to_chat(world, "<b>Error in setup for [GLOB.master_mode].</b> Reverting to pre-game lobby.")
+		job_master.ResetOccupations()
+		return FALSE
+
+	callHook("roundstart")
+
+	CHECK_TICK
+	if(hide_mode)
+		to_chat(world, "<span class='infoplain'><B>The current game mode is - Secret!</B></span>")
+	else
+		mode.announce()
+
+	if(CONFIG_GET(flag/ooc_allowed))
+		config.ooc_allowed = TRUE
+
+	CHECK_TICK
+	//Here we will trigger the auto-observe and auto bst debug things
+	if (CONFIG_GET(flag/auto_observe))
+		for(var/client/C in GLOB.clients)
+			if (C.mob)
+				make_observer(C.mob)
+
+	CHECK_TICK
+	if (CONFIG_GET(flag/auto_bst))
+		for(var/client/C in GLOB.clients)
+			if (C.mob)
+				C.cmd_dev_bst(TRUE)
+
+	if (CONFIG_GET(flag/debug_verbs))
+		for(var/client/C in GLOB.clients)
+			C.enable_debug_verbs(TRUE)
+
+	CHECK_TICK
+	for(var/I in round_start_events)
+		var/datum/callback/cb = I
+		cb.InvokeAsync()
+	LAZYCLEARLIST(round_start_events)
+	CHECK_TICK
+
+	log_world("Game start took [(world.timeofday - init_start) / 10]s")
+	round_start_time = world.time
+
+	current_state = GAME_STATE_PLAYING
+	Master.SetRunLevel(RUNLEVEL_GAME)
+
+	CHECK_TICK
+	PostSetup()
+
+	if(CONFIG_GET(flag/sql_enabled))
+		statistic_cycle() // Polls population totals regularly and stores them in an SQL DB -- TLE
+
+	return TRUE
+
+
+/datum/controller/subsystem/ticker/proc/PostSetup()
+	set waitfor = FALSE
+	mode.post_setup()
+
+	//Holiday Round-start stuff	~Carn
+	Holiday_Game_Start()
+
+	setup_done = TRUE
+
+
+//These callbacks will fire after roundstart key transfer
+/datum/controller/subsystem/ticker/proc/OnRoundstart(datum/callback/cb)
+	if(!HasRoundStarted())
+		LAZYADD(round_start_events, cb)
+	else
+		cb.InvokeAsync()
+
+
+//These callbacks will fire before roundend report
+/datum/controller/subsystem/ticker/proc/OnRoundend(datum/callback/cb)
+	if(current_state >= GAME_STATE_FINISHED)
+		cb.InvokeAsync()
+	else
+		LAZYADD(round_end_events, cb)
+
+
+/datum/controller/subsystem/ticker/proc/station_explosion_cinematic(station_missed=0, override = null)
+	if(cinematic)
+		return	//already a cinematic in progress!
+
+	//initialise our cinematic screen object
+	cinematic = new(src)
+	cinematic.icon = 'icons/effects/station_explosion.dmi'
+	cinematic.icon_state = "station_intact"
+	cinematic.plane = HUD_PLANE
+	cinematic.layer = HUD_ABOVE_ITEM_LAYER
+	cinematic.mouse_opacity = 0
+	cinematic.screen_loc = "1,0"
+
+	var/obj/structure/bed/temp_buckle = new(src)
```

Comment:
this is atrociously bad

Reply:
Deleted and added to do with rework buckled.